message/validation/ratelimiter/peer_limiter.go

```diff
@@ -0,0 +1,124 @@
+package ratelimiter
+
+import (
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/time/rate"
+)
+
+type Config struct {
+	// InvalidRSAs is the number of invalid RSA messages allowed per second
+	InvalidRSAs int
+
+	// InvalidMessages is the number of invalid messages allowed per second
+	InvalidMessages int
+
+	// BlockDuration is the duration for which a peer is blocked after exceeding the rate limit
+	BlockDuration time.Duration
+
+	// CacheSize is the number of peers to keep in the cache
+	CacheSize int
+}
+
+func DefaultConfig() Config {
+	return Config{
+		InvalidRSAs:     5,
+		InvalidMessages: 20,
```

Comment:
InvalidMessages are ignored messages? If so maybe give a clear name
it should be well above 20 for sure because we can have lots of them

I remember we set invalid RSAs to be 20, because IIRC correctly after 20 we also make the peer greylisted on libp2p. why did this change?

Reply:
btw i didn't use the correct numbers here, we'll revert it back to 20 RSAs and 300k messages